schemas/v3.0/schema.yaml

```diff
@@ -756,540 +617,164 @@ definitions:
+          style:
+            enum: [simple]
+            default: simple
+
+      - description: Parameter in cookie
+        properties:
+          in:
+            enum: [cookie]
+          style:
+            enum: [form]
+            default: form
+
+
+  RequestBody:
+    type: object
+    required:
+      - content
     properties:
-      name:
-        type: string
-      in:
-        type: string
-        enum:
-          - query
       description:
         type: string
+      content:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/MediaType'
       required:
         type: boolean
         default: false
-      deprecated:
-        type: boolean
-        default: false
-      allowEmptyValue:
-        type: boolean
-        default: false
-      style:
-        type: string
-        enum:
-          - form
-          - spaceDelimited
-          - pipeDelimited
-          - deepObject
-        default: form
-      explode:
-        type: boolean
-      allowReserved:
-        type: boolean
-        default: false
-      schema:
-        oneOf:
-          - $ref: '#/definitions/Schema'
-          - $ref: '#/definitions/Reference'
-      example: {}
     patternProperties:
+      '^\$ref$':
+        $ref: '#/definitions/RefProperty'
       '^x-': {}
-    additionalProperties: false    
+    additionalProperties: false
 
-  ParameterWithSchemaWithExampleInHeader:
+  SecurityScheme:
+    oneOf:
+      - $ref: '#/definitions/APIKeySecurityScheme'
+      - $ref: '#/definitions/HTTPSecurityScheme'
+      - $ref: '#/definitions/OAuth2SecurityScheme'
+      - $ref: '#/definitions/OpenIdConnectSecurityScheme'
+
+  APIKeySecurityScheme:
     type: object
     required:
+      - type
       - name
       - in
-      - schema
     properties:
+      type:
+        type: string
+        enum:
+          - apiKey
       name:
         type: string
       in:
         type: string
         enum:
           - header
+          - query
+          - cookie
       description:
         type: string
-      required:
-        type: boolean
-        default: false
-      deprecated:
-        type: boolean
-        default: false
-      allowEmptyValue:
-        type: boolean
-        default: false
-      style:
-        type: string
-        enum:
-          - simple
-        default: simple
-      explode:
-        type: boolean
-      allowReserved:
-        type: boolean
-        default: false
-      schema:
-        oneOf:
-          - $ref: '#/definitions/Schema'
-          - $ref: '#/definitions/Reference'
-      example: {}
     patternProperties:
       '^x-': {}
-    additionalProperties: false    
+    additionalProperties: false
 
-  ParameterWithSchemaWithExampleInCookie:
+  HTTPSecurityScheme:
     type: object
     required:
-      - name
-      - in
-      - schema
+      - scheme
+      - type
     properties:
-      name:
+      scheme:
         type: string
-      in:
+        not:
```

Comment:
I'm not sure that your combinations of `not` and `enum` all work out.  

Much farther down, you `oneOf` both bearer and not bearer, but you have not bearer here.  I think you just want `type: string` here without any enums, and only do the enum in the `oneOf`

There are some other `enum`/`oneOf` combinations that look similar that may be worth a closer look.  I have not gone through all in detail.

Reply:
Indeed, this `not enum` should not be there, thank you! I checked other places with `not` or `enum` and they seem to be ok. 

But the best would be to run validation against some large set of OAI schema to see if there is a regression.